acquire-driver-common/src/simcams/simulated.camera.c

```diff
@@ -376,9 +406,25 @@ simcam_set(struct Camera* camera, struct CameraProperties* settings)
     };
 
     size_t nbytes = aligned_bytes_of_image(shape);
-    self->im.data = malloc(nbytes);
-    EXPECT(self->im.data, "Allocation of %llu bytes failed.", nbytes);
+    free(self->im.frame_data);
+
+    self->im.frame_data = malloc(nbytes);
+    if (!self->im.frame_data) {
+        LOGE("Allocation of %llu bytes failed.", nbytes);
+        lock_release(&self->im.lock);
+        goto Error;
+    }
 
+    free(self->im.render_data);
+
+    self->im.render_data = malloc(nbytes);
+    if (!self->im.render_data) {
+        LOGE("Allocation of %llu bytes failed.", nbytes);
+        lock_release(&self->im.lock);
+        goto Error;
+    }
+
+    lock_release(&self->im.lock);
     return Device_Ok;
 Error:
     return Device_Err;
```

Comment:
Probably worth factoring these code blocks into a function. Something like the suggestion below. One caveat is that the failure handling assumes you're inside the lock, so doing this requires modifying the top part of `simcam_set` as well.

```suggestion
    CHECK(checked_realloc(self->im.frame_data,nbytes);
    CHECK(checked_realloc(self->im.render_data,nbytes);
    
    int status=Device_Ok;
Finalize:
    lock_release(&self->im.lock);
    return status;
Error:
    status=Device_Err;
    goto Finalize;
```

where

```c
void* checked_realloc(void* in,size_t nbytes) {
    // if `in` is NULL, realloc is identical to malloc.
    void* out = realloc(in,nbytes);
    EXPECT(out,"Allocation of %llu bytes failed",nbytes);
Error:
    return out;
}
```

Reply:
Alternative:

```c
static void*
checked_realloc(void* in, size_t nbytes)
{
    void* out = realloc(in, nbytes);
    EXPECT(out, "Allocation of %llu bytes failed.", nbytes);

Finalize:
    return out;

Error:
    free(in);
    goto Finalize;
}
```